Integrations/integration-AbuseDB.yml

```diff
@@ -0,0 +1,400 @@
+        total_reports = analysis.get("totalReports")
+        abuse_score = analysis.get("abuseConfidenceScore")
+        dbot_score = 0 if total_reports == 0 else 1 if abuse_score < 20 else 2 if abuse_score < threshold else 3
+        return dbot_score
+
+
+    def createEntry(context, table, dbot_scores, title):
+        entry = {
+            'ContentsFormat': formats['json'],
+            'Type': entryTypes['note'],
+            'Contents': context,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(title, table, removeNull=True),
+            'EntryContext': {'AbuseIPDB(val.IP.Address && val.IP.Address == obj.IP.Address)' : createContext(context, removeNull=True),
+                'DbotScore(val.Vendor && val.Indicator && val.Vendor == obj.Vendor && val.Indicator == obj.Indicator)' :
+                    createContext(dbot_scores, removeNull=True),
+            }
+        }
+        return entry
+
+
+    ''' FUNCTIONS '''
+    def check_ip_command(ip, days=MAX_AGE, verbose=VERBOSE, threshold=THRESHOLD):
+        params = {
+            "ipAddress" : ip,
+            "maxAgeInDays" : days
+        }
+        if verbose:
+            params['verbose'] = "verbose"
+        analysis = http_request("GET", url_suffix=CHECK_CMD, params=params).get("data")
+        return analysis_to_entry(analysis, verbose=verbose, threshold=threshold)
+
+
+    def check_block_command(network, limit, days=MAX_AGE, threshold=THRESHOLD):
+        params = {
+            "network" : network,
+            "maxAgeInDays" : days
+        }
+        analysis = http_request("GET", url_suffix=CHECK_BLOCK_CMD, params=params).get("data").get("reportedAddress")
+        return analysis_to_entry(analysis[:int(limit) if limit.isdigit() else 40], verbose=False, threshold=threshold)
+
+
+    def report_ip_command(ip, categories):
+        params = {
+            "ip" : ip,
+            "categories" : ",".join([CATEGORIES_ID[c] if c in CATEGORIES_ID else c for c in categories.split()])
+        }
+        analysis = http_request("POST", url_suffix=REPORT_CMD, params=params)
+        demisto.results("IP address reported successfully.")
+        return "IP address reported successfully."
+
+
+    def get_blacklist_command(limit, days):
+        params = {
+            'maxAgeInDays' : days,
+            "limit" : limit
+        }
+        analysis = http_request("GET", url_suffix=BLACKLIST_CMD, params=params).get("data")
+        return blacklist_to_entry(analysis)
+
+
+    def test_module():
+        try:
+            check_ip_command(ip=TEST_IP, verbose=False)
+        except Exception as e:
+            LOG(e)
+            return_error(e.message)
+        demisto.results('ok')
+
+
+    try:
+        if demisto.command() == 'test-module':
+            # Tests connectivity and credentails on login
+            test_module()
+        elif demisto.command() == 'ip':
+            demisto.results(check_ip_command(**demisto.args()))
```

Comment:
let's have a short talk about this argument handling.
I am not sure this is the way we want to handle the args in `_commands`